core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -478,19 +496,32 @@ class KafkaApisTest {
       )).andAnswer(
         () => responseCallback.getValue.apply(Map(topicPartition -> Errors.COORDINATOR_LOAD_IN_PROGRESS)))
 
-    EasyMock.expect(requestChannel.sendResponse(EasyMock.capture(capturedResponse)))
+    EasyMock.expect(requestChannel.sendResponse(
```

Comment:
- All tests in this class fail because of a change in the sendResponse API that fails to capture the response here.
  - On trunk, `RequestChannel.sendResponse` is called by `KafkaApis.sendResponse` and the response is captured here correctly.
  - On the KIP-500 branch, `KafkaApis.sendResponse` has been moved into another overloaded definition of the `RequestChannel.sendResponse(RequestChannel.Request, Option[AbstractResponse], Option[Send => Unit])` method which eventually calls the other `RequestChannel.sendResponse(RequestChannel.Response)` method expected here. 
- The tests use a mock of the RequestChannel object and so, the second call is never made and the response is never captured. 
- To fix that, I moved the expected call one frame up for the new sendResponse which accepts an `AbstractResponse` instead of a `RequestChannel.Response`. I made the change to have all the tests expect this new method call and then retain the logic in the `readResponse` method in this Test class to serialize/deserialize the response and test for the necessary API message checks.